cmake/dependencies/zlib.cmake

```diff
@@ -25,6 +25,12 @@ SET(_download_hash
 SET(_install_dir
     ${RV_DEPS_BASE_DIR}/${_target}/install
 )
+
+# This file is pretty close to being ready to use the Standrd macros: (create_lib_bin especially but maybe rv_make_std_lib). One problem is debug names for Libs which is different but there's ways to fix this.
```

Comment:
Zlib can be made to use the new Macros (from Nicolas), less duplicated code and a nice Clean Target.
Moreover, to align with Autodesk teams, we should update to 1.2.13.